storm-core/src/jvm/backtype/storm/scheduler/Cluster.java

```diff
@@ -560,14 +619,29 @@ public void updateAssignedMemoryForTopologyAndSupervisor(Topologies topologies)
         }
     }
 
+    /**
+     * set scheduler status for a topology
+     */
+    private static final Logger LOG = LoggerFactory
+            .getLogger(Cluster.class);
     public void setStatus(String topologyId, String status) {
         this.status.put(topologyId, status);
     }
 
+    /**
+     * Get all schedule statuses
+     */
     public Map<String, String> getStatusMap() {
         return this.status;
     }
 
+    /**
+     * set scheduler status map
+     */
+    public void setStatusMap(Map<String, String> statusMap) {
+        this.status.putAll(statusMap);
+    }
```

Comment:
This is named like a setter method, but it does not set the status map to what is in `statusMap`, it overrides any entries present in both. Suggest we rename it or change it to replace the current map.